### ISSUE TITLE
The training loss in classify_dvsg remains the same

### ISSUE THREAD

**Issue by KoiLiu** [fangwei123456/spikingjelly#216]
Hello, I met a problem when I ran the code in spikingjelly.clock_driven.classify_dvsg.
When I ran this code by the guidance of the Spikingjelly, the training loss always remained the same.
`Mkdir ./logs/T_20_b_16_c_128_SGD_lr_0.0001_CosALR_32.
100%|███████████████████████████████████████████| 73/73 [00:36<00:00,  2.01it/s]
Namespace(T=20, T_max=32, amp=False, b=16, channels=128, cupy=False, data_dir='./dataset', device='cuda:0', epochs=64, gamma=0.1, j=4, lr=0.0001, lr_scheduler='CosALR', momentum=0.9, opt='SGD', out_dir='./logs', resume=None, step_size=32)
epoch=0, train_loss=0.09090909361839294, train_acc=0.08304794520547945, test_loss=0.09090909361839294, test_acc=0.08333333333333333, max_test_acc=0.08333333333333333, total_time=41.57230806350708
100%|███████████████████████████████████████████| 73/73 [00:35<00:00,  2.05it/s]
Namespace(T=20, T_max=32, amp=False, b=16, channels=128, cupy=False, data_dir='./dataset', device='cuda:0', epochs=64, gamma=0.1, j=4, lr=0.0001, lr_scheduler='CosALR', momentum=0.9, opt='SGD', out_dir='./logs', resume=None, step_size=32)
epoch=1, train_loss=0.09090909361839294, train_acc=0.08304794520547945, test_loss=0.09090909361839294, test_acc=0.08333333333333333, max_test_acc=0.08333333333333333, total_time=40.947574615478516
100%|███████████████████████████████████████████| 73/73 [00:35<00:00,  2.06it/s]
Namespace(T=20, T_max=32, amp=False, b=16, channels=128, cupy=False, data_dir='./dataset', device='cuda:0', epochs=64, gamma=0.1, j=4, lr=0.0001, lr_scheduler='CosALR', momentum=0.9, opt='SGD', out_dir='./logs', resume=None, step_size=32)
epoch=2, train_loss=0.09090909361839294, train_acc=0.08304794520547945, test_loss=0.09090909361839294, test_acc=0.08333333333333333, max_test_acc=0.08333333333333333, total_time=40.77522301673889
100%|███████████████████████████████████████████| 73/73 [00:35<00:00,  2.05it/s]`
As shown above, I just changed the T from 16 to 20, and the dataset was already preprocessed by the guidance of Spikingjelly. The network wasn's changed. By the way, I have tried different LR (0.001, 0.0001) but it didn't work. May I ask you the problem I may set incorrectly or something wrong else?

**Comment by fangwei123456** [comment]
Hi, try the latest version of SJ by installing from source codes from github.

**Comment by fangwei123456** [comment]
Try to use Adam with lr=0.001.

**Comment by KoiLiu** [comment]
Thank you for your kind advice. I git the latest version of SJ from github and use Adam with lr=0.001, and it worked! The training_loss declined!  By the way, I met an error when I 'python setup.py install'. But I think this error doesn't matter. More details are as followed:
`error: unable to create file docs/source/_static/tutorials/clock_driven/14_classify_dvsg/T_16_b_16_c_128_Adam_lr_0.001_CosALR_64_amp_cext/dvsg_logs/events.out.tfevents.1618393625.de41f92009cf3011eb0ac59057a81652d2d0-fangw1714-0.14358.0: Filename too long
fatal: unable to checkout working tree
warning: Clone succeeded, but checkout failed.`
And I still want to know why SGD don't work? :-)

Best

**Comment by fangwei123456** [comment]
`Filename too long fatal` is an error from git. You can enable the long filename by `git config --system core.longpaths true`.

> And I still want to know why SGD don't work? :-)

Adam is more stable than SGD. You need to try different learning rates for SGD.
